trailingstoploss_tp.py

```diff
@@ -108,39 +110,67 @@ def upgrade_config(thelogger, cfg):
                 cfg.write(cfgfile)
 
             thelogger.info("Upgraded section %s to have config list" % cfgsection)
-
+        if cfgsection.startswith("tsl_tp_") and cfg.has_option(cfgsection, "config"):
+            jsonconfiglist = list()
+            for configsectionconfig in json.loads(cfg.get(cfgsection, "config")):
+                if "sl-timeout" not in configsectionconfig:
+                    cfgsectionconfig = {
+                        "activation-percentage": configsectionconfig.get("activation-percentage"),
+                        "initial-stoploss-percentage": configsectionconfig.get("initial-stoploss-percentage"),
+                        "sl-timeout": "0",
+                        "sl-increment-factor": configsectionconfig.get("sl-increment-factor"),
+                        "tp-increment-factor": configsectionconfig.get("tp-increment-factor"),
+                    }
+                    jsonconfiglist.append(cfgsectionconfig)
+
+            if len(jsonconfiglist) > 0:
+                cfg.set(cfgsection, "config", json.dumps(jsonconfiglist))
+
+                with open(f"{datadir}/{program}.ini", "w+") as cfgfile:
+                    cfg.write(cfgfile)
+
+                thelogger.info("Upgraded section %s to have config list" % cfgsection)
     return cfg
 
 
-def update_deal(thebot, deal, new_stoploss, new_take_profit):
+def update_deal(thebot, deal, new_stoploss, new_take_profit, sl_timeout):
     """Update bot with new SL and TP."""
 
     bot_name = thebot["name"]
     deal_id = deal["id"]
 
+    payload = {
+        "deal_id": thebot["id"],
+        "stop_loss_percentage": new_stoploss,
+        "take_profit": new_take_profit,
+    }
+
+    # Check whether there is a timeout set for this config, when there is, set the timeout to this value.
+    # Otherwise, send the payload as it is.
+    if sl_timeout != 0:
+        payload["stop_loss_timeout_enabled"] = True
+        payload["stop_loss_timeout_in_seconds"] = sl_timeout
+
     error, data = api.request(
         entity="deals",
         action="update_deal",
         action_id=str(deal_id),
-        payload={
-            "deal_id": thebot["id"],
-            "stop_loss_percentage": new_stoploss,
-            "take_profit": new_take_profit,
-        },
+        payload=payload,
     )
     if data:
         logger.info(
             f"Changing SL for deal {deal['pair']} ({deal_id}) on bot \"{bot_name}\"\n"
             f"Changed SL from {deal['stop_loss_percentage']}% to {new_stoploss}%. "
-            f"Changed TP from {deal['take_profit']}% to {new_take_profit}%"
+            f"Changed TP from {deal['take_profit']}% to {new_take_profit}% "
+            f"Changed SL timeout to {sl_timeout}s."
```

Comment:
Is het niet handig om voor de SL timeout ook een from - to te laten zien? Blijft het consistent met de rest namelijk, ookal zou het kunnen dat de waarden hetzelfde zijn gebleven.